src/Core/src/HttpClient/AwsRetryDecider.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace AsyncAws\Core\HttpClient;
+
+use AsyncAws\Core\AwsError\AwsErrorFactory;
+use AsyncAws\Core\Exception\ParseResponse;
+use Symfony\Component\HttpClient\Retry\RetryDeciderInterface;
+
+/**
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class AwsRetryDecider implements RetryDeciderInterface
+{
+    public function shouldRetry(
+        string $requestMethod,
+        string $requestUrl,
+        array $requestOptions,
+        int $responseStatusCode,
+        array $responseHeaders,
+        ?string $responseContent
+    ): ?bool {
+        if (\in_array($responseStatusCode, [423, 425, 429, 500, 502, 503, 504, 507, 510], true)) {
+            return true;
+        }
+        if (
+            400 !== $responseStatusCode
+            || (int) ($responseHeaders['content-length'][0] ?? '0') > 1024
```

Comment:
Prevent downloading/buffering body when the size is way too much and couldn't be a valid retryable response.
Maybe this value should be increased. (ie. 100KB)

Reply:
I am not sure what the correct size is. Let's just write a comment and leave it with 1kB

Reply:
An XML errors message is about 250char (https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html)

Maybe with bigger message and description, it could be higher that 1 KIB.

I changed it to 16KiB which should be fine IMHO

Reply:
Good. Can you also add a comment about it?